### DISCUSSION TITLE
Optimal clustering paramters

### DISCUSSION THREAD

**Discussion by AnjaliC4** [GreenleafLab/ArchR#1158] [Questions / Documentation]
Hello @rcorces, 

I would like to take a moment to say thank you for such a great tool and it works very well. :)

There seem to be two major ways of identifying clusters - 1) Call major clusters/cell-types and sub-cluster them 2)Call all the clusters with higher resolution. I see that current papers are using several ways to identify "stable clusters" using, for example, bootstrapping approaches, so as to decide the no. of dimensions and best clustering resolution for a dataset. Most of these methods import a seurat/signac object or use an accessibility matrix. Given that ArchR framework is different and creates a large bin accessibility matrix, it is not possible to use those methods for large snATAC-seq datasets. I'm wondering if you can suggest some ways to identify the most optimal LSI (no. of variable peaks/no. of iterations) and clustering parameters to identify clusters in ArchR. I understand that this can be subjective, but moving forward, it can be useful to make more objective decisions when choosing these parameters. Could you please suggest some pipelines or methods to optimize these parameters in ArchR?

Thank you for your advice!

**Comment by rcorces** [comment]
I dont have anything to add here unfortunately. The default parameters in ArchR are out best approximation of optimal for the general user base of ArchR. There are hundreds / thousands of ways to do these things and its not in the scope of what ArchR is doing right now to address this particular problem.

**Reply by AnjaliC4** [reply]
Hi @rcorces, I agree with you. However, exporting 500bp tile matrix from 100s of samples will not be feasible to integrate with the available tools for doing these analyses and choosing optimal clustering parameters. Please allow me to reframe and ask you whether you think there may be a way to compress the tile matrix or convert into other formats for such analysis?

**Reply by rcorces** [reply]
Well, you could limit yourself to performing LSI on the PeakMatrix and then export the peak matrix or use larger tiles. I dont think there is a good way to export the tile matrix. Its hard to give a very satisfying response here. Most of the tools that come to mind are designed for RNA and its way easier to do these things on the scale of RNA data.

**Reply by AnjaliC4** [reply]
Hi @rcorces, it makes sense to use PeakMatrix derived from major clusters to chose optimal resolutions to identify sub-clusters within the major cell-types. I have noticed that in your paper, you do something similar where you sub-cluster the excitatory neuronal cluster. I have a question related to that. There is an option in latest Seurat version to FindSubClusters() within a cluster - is it possible to implement something like that in ArchR with AddSubClusters() to provide different resolutions to sub-cluster every cluster based on the bins or pseudo-bulk peak matrix of every cluster? I think this would really help. If you think this not soemthing ArchR can implement anytime soon - could you suggest how can I see all the sub-clusters in a joint embedding space -after sub-clustering the major cell-types in ArchR. Thank you once again!

**Reply by rcorces** [reply]
> There is an option in latest Seurat version to FindSubClusters() within a cluster - is it possible to implement something like that in ArchR with AddSubClusters() to provide different resolutions to sub-cluster every cluster based on the bins or pseudo-bulk peak matrix of every cluster?

The unfortunate truth is that new feature requests are currently taking a back seat as we look for dedicated funding and hands to continue ArchR development. I would be happy to review a pull request if someone wants to implement this.

> could you suggest how can I see all the sub-clusters in a joint embedding space -after sub-clustering the major cell-types in ArchR

I'm not sure I understand the question. You should be able to visualize whatever cells you want, colored by whatever annotation you want. Using project subsetting you should be able to do any subclustering you desire. Perhaps I'm not interpreting the question correctly.

**Reply by AnjaliC4** [reply]
Hi @rcorces, Thanks for your response. My second question was how to re-embed all the sub-clusters with UMAP. My understanding is that I will need to create a subsetted ArchRproj for every major cell-type - perform LSI & clustering on them individually. however, I do not know fully understand how to then re-embed all the these sub-clusters from different subsetted ArchR projects to one combined UMAP embedding. Or even how to combine these subsetted ArchR projects for analysis across all the sub-clusters. Hope that my question makes more sense now?

**Reply by rcorces** [reply]
I think you are thinking about this problem in the wrong way. You cannot simply re-embed the subclusters. However, you can take the cluster identities learned from sub-clustering and add those to the cells in your original ArchR project in `cellColData`. Then you can plot your original UMAP but colored by a new column that contains the subcluster information.

**Reply by AnjaliC4** [reply]
Ok, thank you @rcorces. That makes a lot of sense! 
I know you guys are very busy, but can I please ask you when you are expecting chapter 17 - plot aesthetics/ other functionalities to be released? Just curious because it would be very useful for my paper. Thanks again for this wonderful tool and for your clarifications.

**Reply by rcorces** [reply]
I would not expect an update on this any time soon. I cannot provide an estimate.
